inc/Ui/DiffView.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace dokuwiki\Ui;
+
+use dokuwiki\ChangeLog\RevisionInfo;
+use dokuwiki\Ui\PageDiff;
+use dokuwiki\Ui\MediaDiff;
+
+/**
+ * DokuWiki Generic Diff Viewer
+ *
+ * @package dokuwiki\Ui
+ */
+class DiffView extends Ui
```

Comment:
Interesting to see this kind of code patterns in commit dc0bfe2177cd4574ccd923cb55e29c5eb0d304e5. I'm not a specialist on these kind of things, so for me its fine...
However, I was wondering why did you introduce it?

Reply:
@Klap-in , this type of refactoring will make clear the class responsibility. I expected you are interested in to make MediaDiff something better in future, so you can focus MediaDiff only without changing other files. Anyway, I withdraw introduce strategy method pattern for the moment.

Reply:
Yeah, indeed, maybe it is for now more practical to ensure first that all functionality is as expected. And move larger code refactors to a bit later moment, the media manager part of the code needs still a lot of attention, that is not quickly solved in this PR ;-) I think it is useful to move now to a stable situation of DokuWiki code, such that a release of DokuWiki can be prepared.

About the code pattern, maybe tuning of the naming can help to make it more transparent. At the moment the naming is quite the same, so for me the responsibilities are not that clear.
An interface is kind of contract, it is not? So maybe a name suggesting what is promised? And as there is a lot of Diff in the name, it can moved to its Ui/Diff/ namespace? But again, I’m not experienced on this, so probably others better can help with useful names?